src/utils/file_manager.py

```diff
@@ -0,0 +1,137 @@
+import json
+from pathlib import Path
+from typing import Optional
+
+import yaml
+
+from constants import SECRETS_YAML, WORK_PREFERENCES_YAML, PLAIN_TEXT_RESUME_YAML
+from src.logging import logger
+
+
+class FileManager:
+    """Class for handling file operations related to the application."""
+
+    @staticmethod
+    def validate_data_folder(app_data_folder: Path) -> tuple:
+        """
+        Validates that the required files exist in the data folder.
+        """
+        if not app_data_folder.exists() or not app_data_folder.is_dir():
+            raise FileNotFoundError(f"Data folder not found: {app_data_folder}")
+
+        required_files = [SECRETS_YAML, WORK_PREFERENCES_YAML, PLAIN_TEXT_RESUME_YAML]
+        missing_files = [file for file in required_files if not (app_data_folder / file).exists()]
+
+        if missing_files:
+            raise FileNotFoundError(
+                f"Missing files in the data folder: {', '.join(missing_files)}"
+            )
+
+        output_folder = app_data_folder / "output"
+        output_folder.mkdir(exist_ok=True)
+        return (
+            app_data_folder / SECRETS_YAML,
+            app_data_folder / WORK_PREFERENCES_YAML,
+            app_data_folder / PLAIN_TEXT_RESUME_YAML,
+            output_folder,
+        )
+
+    @staticmethod
+    def file_paths_to_dict(resume_file: Optional[Path], plain_text_resume_file: Path) -> dict:
+        """
+        Returns a dictionary containing paths to the resume and plain text resume files.
+        """
+        if not plain_text_resume_file.exists():
+            raise FileNotFoundError(
+                f"Plain text resume file not found: {plain_text_resume_file}"
+            )
+
+        result = {"plainTextResume": plain_text_resume_file}
```

Comment:
stick to snake case

Reply:
The dictionary returned by the get_file_paths method was updated to use snake_case keys (plain_text_resume instead of plainTextResume).